sdcm/tester.py

```diff
@@ -654,8 +654,16 @@ def init_resources(self, loader_info=None, db_info=None,
             self.get_cluster_baremetal()
 
         seeds_num = self.params.get('seeds_num', default=1)
-        for i in range(seeds_num):
-            self.db_cluster.nodes[i].is_seed = True
+        if self.params.get('instance_provision') == 'mixed':
+            for i in range(len(self.db_cluster.nodes)):
+                if seeds_num == 0:
+                    break
+                if self.db_cluster.nodes[i].is_spot:
+                    self.db_cluster.nodes[i].is_seed = True
```

Comment:
I don't know, need to dig into and try to understand the use case.

Reply:
if i am not mistaken:
1. init resource initialize the create nodes and after the instances created, mark them as seeds 
2. upon node setup, the get_seed_nodes_by_flag get all nodes marked as seed and configure the approriate ips in scylla.con
3. when wait_for_init starts, it use the get_seed_nodes to get all seed ips from config scylla.yaml  and mark them again. looks this method is used for libvirt clusters, cassandra cluster, to get correct ips from  scylla.yaml. In Aws cluster, it only get the same nodes and mark them again with the same flag. Here we can add some checks if node is already not marked as seed.
4. The only case where could interfere if one node is seed by config. in this case, first node in cluster marked as seed. i change the order of node initialization, so that Spot instances created first. in this case, even the seed=1, the first spot instance will be seed. and on demand nodes will not be a seed node.